internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestStoreMigration.java

```diff
@@ -532,103 +551,11 @@ private Properties createAndPushStore(String clusterName, String storeName) thro
-        "--cluster-src", srcClusterName, "--cluster-dest", destClusterName, "--fabric", FABRIC0 };
-
-    try (ControllerClient destParentControllerClient = new ControllerClient(destClusterName, controllerUrl)) {
-      TestUtils.waitForNonDeterministicAssertion(60, TimeUnit.SECONDS, () -> {
-        AdminTool.main(completeMigration0);
-        // Store discovery should point to the new cluster after completing migration
-        ControllerResponse discoveryResponse = destParentControllerClient.discoverCluster(storeName);
-        Assert.assertEquals(discoveryResponse.getCluster(), destClusterName);
-      });
-    }
-  }
-
-  private void endMigration(String parentControllerUrl, String childControllerUrl, String storeName) throws Exception {
-    String[] endMigration = { "--end-migration", "--url", parentControllerUrl, "--store", storeName, "--cluster-src",
-        srcClusterName, "--cluster-dest", destClusterName };
-    AdminTool.main(endMigration);
-
-    try (ControllerClient srcControllerClient = new ControllerClient(srcClusterName, parentControllerUrl);
-        ControllerClient destControllerClient = new ControllerClient(destClusterName, parentControllerUrl)) {
-      TestUtils.waitForNonDeterministicAssertion(30, TimeUnit.SECONDS, () -> {
-        // Store should be deleted in source cluster. Store in destination cluster should not be migrating.
-        StoreResponse storeResponse = srcControllerClient.getStore(storeName);
-        Assert.assertNull(storeResponse.getStore());
-
-        storeResponse = destControllerClient.getStore(storeName);
-        Assert.assertNotNull(storeResponse.getStore());
-        Assert.assertFalse(storeResponse.getStore().isMigrating());
-        Assert.assertFalse(storeResponse.getStore().isMigrationDuplicateStore());
-      });
-    }
-    if (childControllerUrl == null) {
-      return;
-    }
-
-    // Perform the same check on child controller too
-    try (ControllerClient srcControllerClient = new ControllerClient(srcClusterName, childControllerUrl);
-        ControllerClient destControllerClient = new ControllerClient(destClusterName, childControllerUrl)) {
-      TestUtils.waitForNonDeterministicAssertion(30, TimeUnit.SECONDS, () -> {
-        // Store should be deleted in source cluster. Store in destination cluster should not be migrating.
-        StoreResponse storeResponse = srcControllerClient.getStore(storeName);
-        Assert.assertNull(storeResponse.getStore());
-
-        storeResponse = destControllerClient.getStore(storeName);
-        Assert.assertNotNull(storeResponse.getStore());
-        Assert.assertFalse(storeResponse.getStore().isMigrating());
-        Assert.assertFalse(storeResponse.getStore().isMigrationDuplicateStore());
-      });
-    }
-
-  }
-
   private void readFromStore(AvroGenericStoreClient<String, Object> client) {
     int key = ThreadLocalRandom.current().nextInt(RECORD_COUNT) + 1;
     client.get(Integer.toString(key));
```

Comment:
A bit surprised to see this, but shouldn't we get the result of the future and assert on it in case there are exceptions?

Reply:
Wow, didn't realize it doesn't get on the future. Will update it and see how it looks. like

Reply:
I've updated it. It's a bit tricky to intentionally catch the exception from the client, not the `RequestBasedMetadata` so didn't add it but add one additional call

Reply:
The client should never throw any exceptions since we are trying to make store migration transparent to our users. The underlying request metadata fetch can fail and do the re-discovery but that shouldn't impact actual read traffic from the client.